pos_self_service_base/static/src/js/screens.js

```diff
@@ -0,0 +1,130 @@
+odoo.define("pos_self_service_base.screens", function(require) {
+    "use strict";
+    // This file contains the base screen where user actions will be included.
+    // Those actions will be defined in seperated modules.
+    // e.g.: `pos_self_service_printing` defines UI and business logic for printing labels,
+
+    var gui = require("point_of_sale.gui");
+    var screens = require("point_of_sale.screens");
+    var core = require("web.core");
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    /* -------- The Self-Service Screen  -------- */
+    // This is the home screen with the call to action buttons.
+    var SelfServiceScreenWidget = screens.ScreenWidget.extend({
+        template: "SelfServiceScreenWidget",
+        scale_weight: 0,
+
+        // Ignore products, discounts, and client barcodes
+        barcode_product_action: function(code) {},
+        barcode_discount_action: function(code) {},
+        barcode_client_action: function(code) {},
+        start: function() {
+            var self = this;
+            this._super();
+            this.self_service_scale_widget = this.pos.chrome.widget.self_service_scale_widget;
+            this.self_service_scale_widget.add_observer(this);
+            this.barcode_parser = this.pos.barcode_reader.barcode_parser;
+        },
+        show: function() {
+            var self = this;
+            this._super();
+            this.chrome.widget.order_selector.hide();
+            this.$(".tare").click(function() {
+                self.click_print();
+            });
+            this.render_button();
+        },
+        get_tare_button_render_env: function() {
+            return {
+                widget: this,
+                pos: this.pos,
+                scale_weight: this.scale_weight,
+            };
+        },
+        update: function(data) {
+            this.scale_weight = data;
+            this.render_button();
+        },
+        renderElement() {
+            var self = this;
+            this._super();
+        },
+        click_print: function() {
+            if (this.scale_weight <= 0) {
+                this.gui.show_popup("alert", {
+                    title: _t("No Weight"),
+                    body: _t("Please put your container on the scale"),
+                });
+            }
+        },
+        format_barcode: function(weight) {
```

Comment:
I think that all the algorithm, could be moved in a separate module "pos_generate_barcode" to make this module more light, and because it is a generic feature. (that should be in the core, by the way in the module ``barcodes``, near the ``barcode_parser.js`` file)

I can make a PR for that point if you want.


some thing that could replace line 62 to line 110 by : 

```
get_barcode: function() {
  return new_function_to_do("tare", this.scale_weight * 1e3);
}

```

What do you think ?

Reply:
Yep, that'll be great. I can give it a try as well :)

Reply:
That's a great idea and gladly accept your proposition regarding the PR :)